deployer/src/deployer/upload.py

```diff
@@ -1,466 +1,401 @@
-            max_bar_width = shutil.get_terminal_size((80, 20)).columns - 30
-            bar_width = int(max_bar_width * done / total_todo)
-            print(
-                f"{done:,} of {total_todo:,}".ljust(20)
-                + f"[{'▋' * bar_width:<{max_bar_width}}] "
-                f"{percentage:.1f}%\r",
-                end="",
-            )
-
-    total_count = 0
-    batch = []
-
-    if config["no_progressbar"]:
-        log = info
-    else:
-        current_log_file_name = Path("upload.log")
-        if current_log_file_name.exists():
-            warning(
-                f"Logging file {current_log_file_name} already exists. "
-                "Will append to it."
-            )
-        info(f"Logging progress into {current_log_file_name}")
-
-        def log(line):
-            with open(current_log_file_name, "a") as f:
-                f.write(f"{line}\n")
-
-    T0 = time.time()
-    for fp in pwalk(directory):
-        if is_junk_file(fp):
-            ignored += 1
-            continue
-        if fp.name.startswith("_"):
-            ignored += 1
-            continue
-        # This assumes  that it can saved in S3 as a key that is the filename.
-        key_path = fp.relative_to(directory)
-        # if key_path.name == "index.redirect":
-        #     # Call these index.html when they go into S3
-        #     key_path = key_path.parent / "index.html"
-        key = f"{config['name']}/{key_path}"
-
-        size = fp.stat().st_size
-        # with open(fp, "rb") as f:
-        #     file_hash = hashlib.md5(f.read()).hexdigest()
-        task = UploadTask(key, fp, size, None, False)
-        if key not in uploaded_already or uploaded_already[key]["Size"] != size:
-            # No doubt! We definitely didn't have this before or it's definitely
-            # different.
-            batch.append(task)
+        return result
 
-        else:
-            # At this point, the key exists and the size hasn't changed.
-            # However, for some files, that's not conclusive.
-            # Image, a 'index.html' file might have this as its diff:
-            #
-            #    - <script src=/foo.a9bef19a0.js></script>
-            #    + <script src=/foo.3e98ca01d.js></script>
-            #
-            # ...which means it definitely has changed but the file size is
-            # exactly the same as before.
-            # If this is the case, we're going to *maybe* upload it.
-            # However, for files that are already digest hashed, we don't need
-            # to bother checking.
-            if _has_hashed_filename(key):
-                skipped += 1
+    def iter_file_tasks(self, build_directory, for_counting_only=False):
+        # Walk the build_directory and yield file upload tasks.
+        for fp in iterdir(build_directory):
+            # Exclude any files that aren't artifacts of the build.
+            if (fp.name == ".DS_Store") or fp.name.endswith("~"):
```

Comment:
```suggestion
            if fp.name.startswith(".") or fp.name.endswith("~"):
```
because there's never a good reason to upload a hidden file.